pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -63,8 +63,8 @@ type LaunchTemplateProvider struct {
 	clusterEndpoint       string
 }
 
-func NewLaunchTemplateProvider(ctx context.Context, ec2api ec2iface.EC2API, amiFamily *amifamily.Resolver, securityGroupProvider *securitygroup.Provider, caBundle *string, startAsync <-chan struct{}, kubeDNSIP net.IP, clusterEndpoint string) *LaunchTemplateProvider {
-	l := &LaunchTemplateProvider{
+func NewProvider(ctx context.Context, ec2api ec2iface.EC2API, amiFamily *amifamily.Resolver, securityGroupProvider *securitygroup.Provider, caBundle *string, startAsync <-chan struct{}, kubeDNSIP net.IP, clusterEndpoint string) *Provider {
+	l := &Provider{
```

Comment:
I think again, we should make an AI, to have the cache hydration be made through a controller mechanism, rather than through a startAsync custom flow

Reply:
It sounds like this has been discussed before. Could you make an issue for this AI?

Reply:
Good call. Done: https://github.com/aws/karpenter/issues/3459